src/equations/equations.jl

```diff
@@ -175,6 +175,10 @@ end
   return flux
 end
 
+# Imposing no boundary condition just evaluates the flux at the inner state.
+@inline function no_boundary_condition(flux, other_args...)
```

Comment:
Agreed. In terms of naming, the first part should be fixed by `boundary_condition_do_nothing`. 

> it is only applicable to flux-type (von Neumann?) BCs (at least as far as I can tell), and thus is too generically named.

It is also applicable to the linear advection outflow "BCs", as well as something like a supersonic outflow, so I think this supports the new name suggested by @ranocha.

Reply:
What about something like `boundary_condition_use_inner_state` or something similar? It would be longer, but also more descriptive in what it does. If I'd read this in an elixir, `boundary_condition_use_inner_state` would immediately give me an idea of what's going on, while `boundary_condition_do_nothing` is less clear. But that's a subjective assessment, of course 😬

Reply:
I think `boundary_condition_do_nothing` would be less intimidating to someone who is unfamiliar with DG, but I agree `boundary_condition_use_inner_state` would be more precise (and maybe less intimidating with an appropriate docstring). 

Thoughts @ranocha?

Reply:
I don't have a strong opinion on this. "do nothing" boundary conditions are well-known for a range of applications, e.g., (incompressible) Navier-Stokes. Since it's just a naming thing, add it to the list in #1147.